Exercises.IEnumerable2/Dependences/ItemInDepthCollection.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace Exercises.IEnumerable2.Dependences
+{
+    public class ItemInDepthCollection : IEnumerable<Item>, IEnumerator<Item>
```

Comment:
Can you organize your code so that `ItemInDepthCollection` cannot be created explicitly in client code (treat your own code as library, and someone who uses your library - writes client code)
Try to limit `new ItemInDepthCollection()` usage or `ItemInDepthCollection` type discovery at all